### PR TITLE
Add: Get edge data function in python graph interface

### PR DESCRIPTION
**Pull request by davvard**

### CODE REVIEW

File: python/networkx.cpp [hunk 1]
```diff
@@ -686,9 +686,17 @@ void ukv::wrap_networkx(py::module& m) {
...
+            auto edge_ids = edges.edge_ids.immutable();
+            auto attrs = read_attributes(g.relations_attrs, edge_ids, name.size() ? name.c_str() : nullptr);
+            if (attrs[0] && attrs[0].size())
+                return py::reinterpret_steal<py::object>(from_json(json_t::parse(attrs[0])));
+            return py::reinterpret_steal<py::object>(py::dict());
+        },
         py::arg("u"),
-        py::arg("v"));
+        py::arg("v"),
+        py::arg("name") = "");
```
Comment (ashvardanian) [inline python/networkx.cpp:699]: This argument name doesn't match NetworkX: https://networkx.org/documentation/stable/reference/classes/generated/networkx.MultiDiGraph.get_edge_data.html

File: python/networkx.cpp [hunk 2]
```diff
@@ -686,9 +686,24 @@ void ukv::wrap_networkx(py::module& m) {
         py::arg("key"));
     g.def(
         "get_edge_data",
-        [](py_graph_t& g, ukv_key_t v1, ukv_key_t v2) { throw_not_implemented(); },
+        [](py_graph_t& g, ukv_key_t v1, ukv_key_t v2, py::object default_value) {
+            std::string default_value_str;
+            to_string(default_value.ptr(), default_value_str);
+            auto edges = g.ref().edges_between(v1, v2).throw_or_release();
+            if (!edges.size())
+                return py::reinterpret_steal<py::object>(Py_None);
```
Comment (ashvardanian) [inline python/networkx.cpp:694]: Why would you "steal" `Py_None`? That looks dangerous. A simple Google Search suggested at [least two strictly better variants from their documentation](https://github.com/pybind/pybind11/issues/1483).

```python
return py::none();
return py::cast<py::none>(Py_None);
```

Comment (ashvardanian) [reply python/networkx.cpp]: So they do work. And it's not the only inclusion in that file. Please check the remaining lines.

I can see why you are using `py::reinterpret_steal` to cast to `py::object`. Do you still need it if you explicitly mention the lambdas return type as `py::object`?

Comment (davvard) [reply python/networkx.cpp]: Yes that's the solution!

**Review by ashvardanian — APPROVED** [review]

**Pull request merged**